src/wmtk/operations/AttributesUpdateBase.cpp

```diff
@@ -1,74 +1,42 @@
 #include "AttributesUpdateBase.hpp"
 
-#include <wmtk/SimplicialComplex.hpp>
-#include <wmtk/TriMesh.hpp>
-#include <wmtk/operations/utils/UpdateVertexMultiMeshMapHash.hpp>
-namespace wmtk::operations {
+#include <wmtk/simplex/closed_star.hpp>
 
-void OperationSettings<AttributesUpdateBase>::create_invariants()
-{
-    invariants = std::make_shared<InvariantCollection>(m_mesh);
-}
-
-AttributesUpdateBase::AttributesUpdateBase(
-    Mesh& m,
-    const Simplex& t,
-    const OperationSettings<AttributesUpdateBase>& settings)
-    : TupleOperation(settings.invariants, t)
-    , m_settings{settings}
-    , m_mesh(m)
-    , m_hash_accessor(get_hash_accessor(m))
-{
-    assert(t.primitive_type() == primitive_type());
-    assert(m.is_valid_slow(t.tuple()));
-    assert(m.is_valid_slow(input_tuple()));
-}
-
-Accessor<long>& AttributesUpdateBase::hash_accessor()
-{
-    return m_hash_accessor;
-}
-
-std::string AttributesUpdateBase::name() const
-{
-    return "attributes_update";
-}
+namespace wmtk::operations {
 
-const Tuple& AttributesUpdateBase::return_tuple() const
-{
-    return m_output_tuple;
-}
 
-std::vector<Simplex> AttributesUpdateBase::modified_primitives() const
-{
-    return {Simplex(primitive_type(), m_output_tuple)};
-}
+AttributesUpdateBase::AttributesUpdateBase(Mesh& m)
+    : Operation(m)
+{}
 
-std::vector<Simplex> AttributesUpdateBase::unmodified_primitives() const
+std::vector<Simplex> AttributesUpdateBase::unmodified_primitives(const Simplex& simplex) const
 {
-    return {input_simplex()};
+    return {simplex};
 }
 
 
-bool AttributesUpdateBase::execute()
+std::vector<Simplex> AttributesUpdateBase::execute(const Simplex& simplex)
 {
-    const SimplicialComplex star = SimplicialComplex::closed_star(mesh(), input_simplex());
-    const auto star_faces = star.get_faces();
-    std::vector<Tuple> incident_face_tuple;
-    incident_face_tuple.reserve(star_faces.size());
-    for (const Simplex& s : star_faces) {
-        incident_face_tuple.emplace_back(s.tuple());
-    }
+    Accessor<long> accessor = hash_accessor();
+    assert(simplex.primitive_type() == primitive_type());
+    assert(mesh().is_valid(simplex.tuple(), accessor));
 
-    mesh().update_vertex_operation_hashes(input_tuple(), hash_accessor());
+    const simplex::SimplexCollection star = simplex::closed_star(mesh(), simplex);
+    const auto star_faces = star.simplex_vector();
```

Comment:
This code does not do the same thing as the old one but I think it is not even used later on so you can just remove it.